### PR TITLE
fix: rtl horizontal scroll view

### LINKED ISSUE #776: [Bug] HorizontalScrollSection doesn't work in RTL
[REMEMBER: a bug should be understandable years later by someone who will never talk to the reporter or assignee.]

STEPS TO REPRODUCE:
1. Go to homepage
2. Select Arabic to trigger RTL layout
3. Observe Latest listings and Neighborhood sections

EXPECTED RESULTS:
Scroll buttons should be in the correct order and should still work correctly to scroll. The leading padding on the first element should line up with the right edge.

OBSERVED RESULTS:
The scroll buttons are flipped incorrectly and don't correctly disable/enable when reaching the end. The leading padding for the first element is on the wrong side so it doesn't line up correctly.

image

ADDITIONAL INFORMATION:

### PR DESCRIPTION
**Pull request by plunkettgoogle**
## Issue

- Closes #776
- Addresses #776, #672

## Description

Please include a summary of the change and which issue(s) is addressed. Please also include relevant motivation and context. List any dependencies that are required for this change. I also took the liberty of adding the translation strings for the titles and subtitles.

### Desktop Arabic:
localhost_3000_ar

### Mobile Arabic:
localhost_3000_ar (1)

### Desktop English:
localhost_3000_ar (3)

### Mobile English:
localhost_3000_ar (2)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/addresses technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Navigate to home page, select Arabic locale.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes
- [ ] I have exported any new pieces in ui-components

### CODE REVIEW

**Comment by plunkettgoogle** [comment]
Oops, just realized you are already addressing the translations in #777. Once that's in, I'll rebase.

File: sites/public/lib/HorizontalScrollSection.tsx [hunk 1]
```diff
@@ -12,27 +12,37 @@ export interface HorizontalScrollSectionProps {
   icon?: IconTypes
 }
 
+function isRtl(): boolean {
```
Comment (GabeBenjamin) [inline sites/public/lib/HorizontalScrollSection.tsx:15]: It seems like the component isn't getting updated when the language changes. I think you might need to use a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) to watch for when document.body.direction changes? My guess is you probably want to add that to useEffect and clean it up as well

**Review by GabeBenjamin — CHANGES_REQUESTED** [review]

File: sites/public/lib/HorizontalScrollSection.tsx (hunk 1 above)
Comment (plunkettgoogle) [reply sites/public/lib/HorizontalScrollSection.tsx:15]: Good catch, I was only manually testing when the view was already scrolled a little bit, so both buttons were already enabled.

**Review by GabeBenjamin — APPROVED** [review]
Nicely done!

**Pull request merged**